libs/as-sdk/assembly/oracle-program.ts

```diff
@@ -0,0 +1,33 @@
+import { Console } from './console';
+import Process from './process';
+
+/**
+ * A class meant to be extended with the custom implementation of an Oracle Program.
+ */
+export class OracleProgram {
+  /**
+   * Meant to be overridden if the oracle program needs to execute logic in the execution phase.
+   */
+  execution(): void {
+    Console.debug('execution not implemented');
```

Comment:
As default we should maybe exit with an exit code of 1 (Since the program was not intended to be used like this) Same with tally